crates/cargo-test-support/src/lib.rs

```diff
@@ -110,7 +110,9 @@ impl FileBuilder {
 
     fn mk(&mut self) {
         if self.executable {
-            self.path.set_extension(env::consts::EXE_EXTENSION);
+            let mut path = self.path.clone().into_os_string();
+            write!(path, "{}", env::consts::EXE_SUFFIX).unwrap();
+            self.path = path.into();
```

Comment:
Why did this change?

Reply:
The goal of the code is to add `.exe` when needed.

Before, we did that by calling `set_extension` but that replaces the extension, instead of adding it.  This blocks us from being able to test precedence between a `./files.rs` and `cargo-file.rs` when running `cargo. file.rs`